Gu.Wpf.DataGrid2D/Cell.cs

```diff
@@ -93,9 +93,33 @@ public static DataTemplateSelector GetEditingTemplateSelector(DependencyObject e
         private static void OnTemplateChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
         {
             var dataGrid = (DataGrid)d;
-            foreach (var column in dataGrid.Columns.OfType<CellTemplateColumn>())
+            if (e.OldValue != null && e.NewValue == null)
+            {
+                for (int i = 0; i < dataGrid.Columns.Count; ++i)
+                {
+                    CellTemplateColumn col = dataGrid.Columns[i] as CellTemplateColumn;
```

Comment:
If you use `as`you should nullcheck after. If you don't nullcheck a direct cast is much better as it throws a more informative exception.
`as` + nullcheck is probably best here.